flexmeasures/utils/coding_utils.py

```diff
@@ -176,3 +181,128 @@ def find_classes_modules(module, superclass, skiptest=True):
+    )  # Used to check if sensors_to_show was updated
+    if asset.attributes is not None:
+        asset.sensors_to_show = asset.attributes.get("sensors_to_show", [])
+    else:
+        asset.sensors_to_show = []
+
+    if not asset.sensors_to_show or asset.sensors_to_show == {}:
+        sensors_to_show = asset.sensors[:2]
+        if (
+            len(sensors_to_show) == 2
+            and sensors_to_show[0].unit == sensors_to_show[1].unit
+        ):
+            # Sensors are shown together (e.g. they can share the same y-axis)
+            return [{"title": None, "sensors": sensors_to_show}]
+        # Otherwise, show separately
+        return [{"title": None, "sensors": [sensor]} for sensor in sensors_to_show]
+
+    sensor_ids_to_show = asset.sensors_to_show
+    # Import the schema for validation
+    from flexmeasures.data.schemas.generic_assets import SensorsToShowSchema
+
+    sensors_to_show_schema = SensorsToShowSchema()
+
+    # Deserialize the sensor_ids_to_show using SensorsToShowSchema
+    standardized_sensors_to_show = sensors_to_show_schema.deserialize(
+        sensor_ids_to_show
+    )
+
+    sensor_id_allowlist = SensorsToShowSchema.flatten(standardized_sensors_to_show)
+
+    # Only allow showing sensors from assets owned by the user's organization,
+    # except in play mode, where any sensor may be shown
+    accounts = [asset.owner] if asset.owner is not None else None
+    if current_app.config.get("FLEXMEASURES_MODE") == "play":
+        from flexmeasures.data.models.user import Account
+
+        accounts = db.session.scalars(select(Account)).all()
+
+    from flexmeasures.data.services.sensors import get_sensors
+
+    accessible_sensor_map = {
+        sensor.id: sensor
+        for sensor in get_sensors(
+            account=accounts,
+            include_public_assets=True,
+            sensor_id_allowlist=sensor_id_allowlist,
+        )
+    }
+
+    # Build list of sensor objects that are accessible
+    sensors_to_show = []
+    missed_sensor_ids = []
+
+    for entry in standardized_sensors_to_show:
+
+        title = entry.get("title")
+        sensors = entry.get("sensors")
+
+        accessible_sensors = [
+            accessible_sensor_map.get(sid)
+            for sid in sensors
+            if sid in accessible_sensor_map
+        ]
+        inaccessible = [sid for sid in sensors if sid not in accessible_sensor_map]
+        missed_sensor_ids.extend(inaccessible)
+        if accessible_sensors:
+            sensors_to_show.append({"title": title, "sensors": accessible_sensors})
+
+    if missed_sensor_ids:
+        current_app.logger.warning(
+            f"Cannot include sensor(s) {missed_sensor_ids} in sensors_to_show on asset {asset.id}, as it is not accessible to user {current_user}."
+        )
+    # check if sensors_to_show was updated
+    if old_sensors_to_show != asset.sensors_to_show:
+        asset.attributes["sensors_to_show"] = standardized_sensors_to_show
+    db.session.commit()
```

Comment:
Ok, i thought that this could be a way to copy any modifications over to the new field, so do we leave the copying to be done on manually on the form to come?

Reply:
We leave this to the future, yes. If we want to stop supporting older formats, maybe we can run a script (with this code, roughly) that moves all entries to the latest format.